src/Process/PostDeploy/WarmUp/Urls.php

```diff
@@ -0,0 +1,175 @@
+        $baseUrl = rtrim($this->urlManager->getBaseUrl(), '/');
+        $urls = [];
+
+        $urlPattern = sprintf(
+            '/^(%s|%s):(\d+|\*):.{1,}/',
+            self::ENTITY_CATEGORY,
+            self::ENTITY_CMS_PAGE
+        );
+
+        foreach ($pages as $page) {
+            if (preg_match($urlPattern, $page)) {
+                $patternUrls = $this->getUrlsByPattern($page);
+                $this->logger->info(sprintf('Found %d urls for pattern "%s"', count($patternUrls), $page));
+                $urls = array_merge($urls, $patternUrls);
+            } else if (strpos($page, 'http') === 0) {
+                if (!$this->isRelatedDomain($page)) {
+                    $this->logger->error(
+                        sprintf(
+                            'Page "%s" can\'t be warmed-up because such domain ' .
+                            'is not registered in current Magento installation',
+                            $page
+                        )
+                    );
+                } else {
+                    $urls[] = $page;
+                }
+            } else {
+                $urls[] = $baseUrl . '/' . $page;
+            }
+        }
+
+        return $urls;
+    }
+
+    /**
+     * Checks that host from $url is using in current Magento installation
+     *
+     * @param string $url
+     * @return bool
+     */
+    private function isRelatedDomain(string $url): bool
+    {
+        if ($this->baseHosts === null) {
+            $this->baseHosts = array_map(
+                function ($baseHostUrl) {
+                    return parse_url($baseHostUrl, PHP_URL_HOST);
+                },
+                $this->urlManager->getBaseUrls()
+            );
+        }
+
+        return in_array(parse_url($url, PHP_URL_HOST), $this->baseHosts);
+    }
+
+    /**
+     * Fetch urls from config:show:urls command and filtering the by given pattern
+     *
+     * @param string $warmUpPattern
+     * @return array
+     */
+    private function getUrlsByPattern(string $warmUpPattern): array
+    {
+        try {
+            list($entity, $storeId, $pattern) = explode(':', $warmUpPattern);
+
+            $command = sprintf('config:show:urls --entity-type="%s"', $entity);
+            if ($storeId && $storeId !== '*') {
+                $command .= sprintf(' --store_id="%s"', $storeId);
+            }
+
+            $process = $this->shell->execute($command);
+
+            $urls = json_decode($process->getOutput());
+
+            if (json_last_error() !== JSON_ERROR_NONE) {
+                $this->logger->error(sprintf(
+                    'Can\'t parse result from command %s: %s',
+                    $command,
+                    json_last_error_msg()
+                ));
+                return [];
+            }
+
+            if ($pattern === '*') {
+                return $urls;
+            }
+
+            $pattern = '/' . str_replace('/', '\/', $pattern) . '/';
```

Comment:
is $pattern a real pattern? will it have all regular expression characters quoted?

Reply:
and the same question about the delimiter. We added to docs not to add it to regexp?

Reply:
in the current implementation, $pattern is a real pattern and it won't be quoted. 
But I'm not sure about the format, maybe is better to create some pseudo-pattern.

Reply:
Added validation for Regex pattern.